qmi/core/transport.py

```diff
@@ -293,6 +293,10 @@ def _parse_keyword_parameters(self, strings: List[str]) -> Dict[str, Any]:
                     ty = self._keywords[k][0]
                     if ty is int and v.startswith('0x'):
                         parameters[k] = int(v, 16)
+                    elif ty == bool:
+                        if v not in ("True", "False"):
+                            raise ValueError()
```

Comment:
The validation of the parameter will be done in the `self._validate...` calls, so this is a double check. Considering that we do need to check the exact string to be correct, it is probably better to do it here, and then in practise we do not need the `validate` method in the `QMI_SerialTransport` class. In principle one _could_ call `QMI_SerialTransport` directly without using our `create_transport`  but this is not instructed in our use of QMI. So, we can drop the second check using the validation method, but it can also stay in case of unintended use. I'm fine with both.